cabal-install/Distribution/Client/ProjectConfig/Types.hs

```diff
@@ -228,6 +232,34 @@ instance Binary ProjectConfigShared
 instance Binary PackageConfig
 
 
+-- | Newtype wrapper for 'Map' that provides a 'Monoid' instance that takes
+-- the last value rather than the first value for overlapping keys.
+newtype MapLast k v = MapLast { getMapLast :: Map k v }
+  deriving (Eq, Show, Functor, Generic, Binary)
+
+instance Ord k => Monoid (MapLast k v) where
+  mempty  = MapLast Map.empty
+  mappend = (<>)
+
+instance Ord k => Semigroup (MapLast k v) where
+  MapLast a <> MapLast b = MapLast (flip Map.union a b)
```

Comment:
yeah... that part was obvious, but why not simply `= MapLast (Map.union b a)`? It's strange to see `flip` used on a saturated function application.